lib/ansible/modules/cloud/amazon/elb_network_lb.py

```diff
@@ -0,0 +1,442 @@
+
+    # Change tags to ansible friendly dict
+    snaked_elb['tags'] = boto3_tag_list_to_ansible_dict(snaked_elb['tags'])
+
+    elb_obj.module.exit_json(changed=elb_obj.changed, **snaked_elb)
+
+
+def delete_elb(elb_obj):
+
+    if elb_obj.elb:
+        elb_obj.delete()
+
+    elb_obj.module.exit_json(changed=elb_obj.changed)
+
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            deletion_protection=dict(type='bool'),
+            listeners=dict(type='list'),
+            name=dict(required=True, type='str'),
+            purge_listeners=dict(default=True, type='bool'),
+            purge_tags=dict(default=True, type='bool'),
+            subnets=dict(type='list'),
+            subnet_mappings=dict(type='list'),
+            scheme=dict(default='internet-facing', choices=['internet-facing', 'internal']),
+            state=dict(choices=['present', 'absent'], type='str'),
+            tags=dict(default={}, type='dict'),
+            wait_timeout=dict(type='int'),
+            wait=dict(type='bool')
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           mutually_exclusive=[['subnets', 'subnet_mappings']])
+
+    # Check for subnets or subnet_mappings if state is present
+    state = module.params.get("state")
+    if state == 'present':
+        if module.params.get("subnets") is None and module.params.get("subnet_mappings") is None:
+            module.fail_json(msg="'subnets' or 'subnet_mappings' is required when state=present")
+
+    # Quick check of listeners parameters
+    listeners = module.params.get("listeners")
+    if listeners is not None:
+        for listener in listeners:
+            for key in listener.keys():
+                if key not in ['Protocol', 'Port', 'DefaultActions']:
+                    module.fail_json(msg="listeners parameter contains invalid dict keys. Should be one of 'Protocol', "
+                                         "'Port', 'DefaultActions'.")
+                # Make sure Port is always an integer
+                elif key == 'Port':
+                    listener[key] = int(listener[key])
+
+                if key == 'Protocol' and listener[key] != 'TCP':
+                    module.fail_json(msg="'Protocol' must be 'TCP'")
+
+            if 'DefaultActions' not in listener.keys():
+                module.fail_json(msg="'DefaultActions' is a required listener dict key")
+
+            if 'Protocol' not in listener.keys():
+                module.fail_json(msg="'Protocol' is a required listener dict key")
+
+            if 'Port' not in listener.keys():
+                module.fail_json(msg="'Port' is a required listener dict key")
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    if region:
```

Comment:
If region is unset it is now handled in boto3_conn so you can remove this conditional.